src/ddmd/mars.d

```diff
@@ -371,7 +371,76 @@ private int tryMain(size_t argc, const(char)** argv)
     updateRealEnvironment(&environment);
     environment.reset(1); // don't need environment cache any more
 
-    parseCommandLine(arguments, argc, global.params, files);
+    if (parseCommandLine(arguments, argc, global.params, files))
+    {
+        Loc loc;
+        errorSupplemental(loc, "run 'dmd -man' to open browser on manual");
+        return EXIT_FAILURE;
+    }
+
+    if (global.params.usage)
+    {
+        usage();
+        return EXIT_SUCCESS;
+    }
+
+    if (global.params.logo)
+    {
+        logo();
+        return EXIT_SUCCESS;
+    }
+
+    if (global.params.mcpuUsage)
+    {
+        printf("
+CPU architectures supported by -mcpu=id:
+  =?             list information on all architecture choices
+  =baseline      use default architecture as determined by target
+  =avx           use AVX 1 instructions
+  =avx2          use AVX 2 instructions
+  =native        use CPU architecture that this compiler is running on
+");
+        return EXIT_SUCCESS;
+    }
+
+    if (global.params.transitionUsage)
+    {
+         printf("
+Language changes listed by -transition=id:
+  =all           list information on all language changes
+  =checkimports  give deprecation messages about 10378 anomalies
+  =complex,14488 list all usages of complex or imaginary types
+  =field,3449    list all non-mutable fields which occupy an object instance
+  =import,10378  revert to single phase name lookup
+  =tls           list all variables going into thread local storage
+");
+        return EXIT_SUCCESS;
+    }
+
+    if (global.params.manual)
+    {
+        version (Windows)
+        {
+            browse("http://dlang.org/dmd-windows.html");
```

Comment:
FYI: https://dlang.org/dmd.html redirects to the corresponding OS automatically, but there is no harm in being explicit here